frontend/integration-tests/views/secrets.view.ts

```diff
@@ -1,12 +1,58 @@
-import { $, $$ } from 'protractor';
+/* eslint-disable no-unused-vars */
+import { $, $$, browser, by, ExpectedConditions as until, element, ElementFinder } from 'protractor';
+
+import * as crudView from '../views/crud.view';
 
 export const secretNameInput = $('#secret-name');
 export const pre = $$('.co-copy-to-clipboard__text');
+export const dt = $$('dl.secret-data dt');
 
 export const saveButton = $('#save-changes');
 export const errorMessage = $('.alert-danger');
+export const authTypeDropdown = $('#dropdown-selectbox');
 
-export const webhookSecretValue = 'webhookValue';
-export const webhookSecretValueInput = $('#webhook-secret-key');
+export const webhookSecretValueInput = $('input[name=webhookSecretKey]');
+export const sourceSecretUsernameInput = $('input[name=username]');
+export const sourceSecretPasswordInput = $('input[name=password]');
+export const sourceSecretSSHTextArea = $('.co-file-dropzone__textarea');
 
 export const createWebhookSecretLink = $('#webhook-link');
+export const createSourceSecretLink = $('#source-link');
+
+export const visitSecretsPage = async(appHost: string, ns: string) => {
+  await browser.get(`${appHost}/k8s/ns/${ns}/secrets`);
+  await crudView.isLoaded();
+};
+
+export const createSecret = async(linkElement: ElementFinder, ns: string, name: string, updateForm: Function) => {
+  await crudView.createItemButton.click();
+  await linkElement.click();
+  await secretNameInput.sendKeys(name);
+  await updateForm();
+  await saveButton.click();
+  expect(errorMessage.isPresent()).toBe(false);
+};
+
+export const checkSecret = async(ns: string, name: string, keyValuesToCheck: Object) => {
+  await browser.wait(until.urlContains(`/k8s/ns/${ns}/secrets/${name}`));
+  await browser.wait(until.textToBePresentInElement($('.co-m-pane__heading'), name));
+  await element(by.partialButtonText('Reveal Values')).click();
+  dt.each(async(el, index) => {
+    const key = await el.getAttribute('textContent');
+    if (keyValuesToCheck[key]) {
+      expect(pre.get(index).getText()).toEqual(keyValuesToCheck[key]);
+    } else {
+      fail();
+    }
+  });
```

Comment:
It looks like this won't fail the test if a key/value pair is missing.

Reply:
yeah, sorry for that